common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -41,34 +44,33 @@
    * @param buffer the byte buffer to be unmapped
    */
   public static void cleanDirectBuffer(ByteBuffer buffer) {
-    if (buffer == null) {
-      return;
-    }
-    if (buffer.isDirect()) {
-      try {
-        if (sCleanerMethod == null) {
-          sCleanerMethod = buffer.getClass().getMethod("cleaner");
-          sCleanerMethod.setAccessible(true);
-        }
-        final Object cleaner = sCleanerMethod.invoke(buffer);
-        if (cleaner != null) {
-          if (sCleanMethod == null) {
-            sCleanMethod = cleaner.getClass().getMethod("clean");
-          }
-          sCleanMethod.invoke(cleaner);
-        }
-      } catch (Exception e) {
-        LOG.warn("Fail to unmap direct buffer due to ", e);
-        buffer = null;
+    Preconditions.checkNotNull(buffer);
+    Preconditions.checkArgument(buffer.isDirect(), "buffer isn't a DirectByteBuffer");
+    try {
+      if (sByteBufferCleanerMethod == null) {
+        sByteBufferCleanerMethod = buffer.getClass().getMethod("cleaner");
+        sByteBufferCleanerMethod.setAccessible(true);
+      }
+      final Object cleaner = sByteBufferCleanerMethod.invoke(buffer);
+      if (cleaner == null) {
+        LOG.error("Failed to get cleaner for ByteBuffer");
+        return;
```

Comment:
Hmm if we are going to do it this way we should probably note that the buffer should not be used after calling this method. The default implementation of the cleaner just marks the allocated memory as unused, but in our case we also destroy the bytebuffer.